cfme/automate/dialog_tab.py

```diff
@@ -0,0 +1,117 @@
+from widgetastic.widget import Text
+from widgetastic_patternfly import Input, Dropdown
+from cached_property import cached_property
+
+from utils.appliance import Navigatable
+from utils.appliance.implementations.ui import navigator, CFMENavigateStep, navigate_to
+from .service_dialogs import AddDialogView
+
+
+class TabForm(AddDialogView):
+    tab_label = Input(name='tab_label')
+    tab_desc = Input(name="tab_description")
+
+
+class AddTabView(TabForm):
+
+    plus_btn = Dropdown('Add')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_customization and self.service_dialogs.is_opened and
+            self.title.text == "Adding a new Dialog [Tab Information]"
+        )
+
+
+class EditTabView(TabForm):
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_customization and self.service_dialogs.is_opened and
+            self.title.text == "Editing Dialog {} [Tab Information]".format(self.tab_label)
+        )
+
+
+class DetailsTabView(TabForm):
+    title = Text("#explorer_title_text")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_customization and self.service_dialogs.is_opened and
+            self.title.text == 'Dialog "{}"'.format(self.context['object'].tab_label)
+        )
+
+
+class TabCollection(Navigatable):
+    def __init__(self, parent):
+        self.parent = parent
+        Navigatable.__init__(self, appliance=parent.appliance)
+
+    @property
+    def tree_path(self):
+        return self.parent.tree_path
+
+    def instantiate(self, tab_label=None, tab_desc=None):
+        return Tab(self,
+            tab_label=tab_label, tab_desc=tab_desc)
+
+    def add_tab(self):
+        view = navigate_to(self, "AddTab")
+        view.plus_btn.item_select("Add a new Tab to this Dialog")
+
+    def create(self, tab_label=None, tab_desc=None):
+        """ Create tab method"""
+        view = navigate_to(self, "AddTab")
+        view.plus_btn.item_select("Add a new Tab to this Dialog")
```

Comment:
I see you have add_tab method which does exactly the same as tehse two lines. You can let the method return the view and then you can use it here instead of navigate_to.